app/webpacker/controllers/product_controller.js

```diff
@@ -0,0 +1,36 @@
+import { Controller } from "stimulus";
+
+// Dynamically update related Product unit fields (expected to move to Variant due to Product Refactor)
+//
+export default class ProductController extends Controller {
+  connect() {
+    // idea: create a helper that includes a nice getter/setter for Rails model attr values, just pass it the attribute name.
+    // It could automatically find (and cache a ref to) each dom element and get/set the values.
+    this.variantUnit = this.element.querySelector('[name$="[variant_unit]"]');
+    this.variantUnitScale = this.element.querySelector('[name$="[variant_unit_scale]"]');
+    this.variantUnitWithScale = this.element.querySelector('[name$="[variant_unit_with_scale]"]');
+
+    // on variant_unit_with_scale changed; update variant_unit and variant_unit_scale
+    this.variantUnitWithScale.addEventListener("change", this.#updateUnitAndScale.bind(this), {
+      passive: true,
```

Comment:
What's this?

Reply:
As in the code comments below: 

```
  // Extract variant_unit and variant_unit_scale from dropdown variant_unit_with_scale,
  // and update hidden product fields
```

![Screenshot 2024-03-25 at 1 54 28 pm](https://github.com/openfoodfoundation/openfoodnetwork/assets/4188088/ff17e1ef-07b9-4a94-9c6a-51d5452713c9)

Reply:
Sorry, I was asking about `passive: true`. I haven't come across that yet.

Reply:
Fair enough, I only recently came across it. It signifies to the browser that it doesn't have to wait for the handler to finish, before proceeding with the default action. 
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#using_passive_listeners
In my interpretation, it means it can run asynchronously, so I've been trying to use it where possible to avoid slowing down the user interface.